src/vm/eventpipefile.h

```diff
@@ -25,37 +26,44 @@ class EventPipeFile : public FastSerializableObject
         );
         ~EventPipeFile();
 
-        // Write an event to the file.
         void WriteEvent(EventPipeEventInstance &instance);
 
-        // Serialize this object.
-        // Not supported - this is the entry object for the trace,
-        // which means that the contents hasn't yet been created.
-        void FastSerialize(FastSerializer *pSerializer)
+        void WriteEnd();
+
+        const char* GetTypeName()
         {
             LIMITED_METHOD_CONTRACT;
-            _ASSERTE(!"This function should never be called!");
+            return "Trace";
         }
 
-        // Get the type name of this object.
-        const char* GetTypeName()
+        void FastSerialize(FastSerializer *pSerializer)
         {
-            LIMITED_METHOD_CONTRACT;
-            return "Microsoft.DotNet.Runtime.EventPipeFile";
+            pSerializer->WriteBuffer((BYTE*)&m_fileOpenSystemTime, sizeof(m_fileOpenSystemTime));
```

Comment:
I like this! Much better that the file itself is represented in FastSerialization instead of being special cased.